app/controllers/carts_controller.rb

```diff
@@ -2,13 +2,14 @@ class CartsController < ApplicationController
 
   def add
     dish = Dish.find(params[:id])
+    @quantity = params[:amount].nil? ? 1 : params[:amount]
```

Comment:
What happens when you set a pry here. What does your params look like?

Reply:
No need for instance variable here. Local will do

Reply:
The amounts are nil, @tochman